server/src/main/java/org/opensearch/index/cache/request/ShardRequestCache.java

```diff
@@ -94,5 +106,15 @@ public void onRemoval(Accountable key, BytesReference value, boolean evicted) {
         if (value != null) {
             dec += value.ramBytesUsed();
         }
+        if ((totalMetric.count() - dec) < 0) {
```

Comment:
Couldn't this version of `onRemoval` just call the other one? (Isn't that true for all of the versions that take `Accountable`?)

Reply:
Yeah we can actually do this. I will change it.